### PR TITLE
Source spoke master fix

### PR DESCRIPTION
**Pull request by jkonecny12**
I want to fix source spoke which was somewhat broken and I found that this change will be somewhat bigger than I expected :).

So in short:
* Change hashing of repositories to special IDs (they are attached to existing RepoData instances)
* Checks can now test even not selected objects
* Check repositories URL before leaving spoke
* Some minor fixes

*This is some kind of RFE so please feel free to tell me if there is better solution*

### CODE REVIEW

**Comment by vpodzime** [comment]
Looks good to me othewise.

**Comment by jkonecny12** [comment]
> I think self._repo_counter should be a generator and we should use the next() method to get the next value and increase the counter with no way to do forget about updating the counter.

I like this idea. I will change it today.

> I think we should either use Python's id() function (if we are really interested in the object ID) or change addon_repo_id to repo_id and add it to pyanaconda.kickstart.RepoData (our "overrides" for kickstart objects).

Yeah I was thinking about it and I don't want to use Python's id() because from time to time the repositories are changing (I don't know why but they are recreated) and recreation change the Python's id(). I think changing the name to `repo_id` is good idea.
About the RepoData I was thinking about this too but I didn't want to add `repo_id` to Pykickstart when It's used only in SourceSpoke. What do you think should this be in pykickstart instead of this injection?

**Comment by vpodzime** [comment]
> About the RepoData I was thinking about this too but I didn't want to add repo_id to Pykickstart when It's used only in SourceSpoke. What do you think should this be in pykickstart instead of this injection?

We have our own ``RepoData`` class in ``pyanaconda.kickstart`` that inherits from the Pykickstart's one. That's the place where this extra attribute should be added and we do the same for some others.

**Comment by jkonecny12** [comment]
Neat!  Thank you for this information. I will upgrade this PR.

**Comment by jkonecny12** [comment]
I have updated the PR I hope it's alright now.

**Comment by jkonecny12** [comment]
I fixed pylint errors

File: pyanaconda/ui/gui/spokes/source.py [hunk 1]
```diff
@@ -403,6 +405,14 @@ def apply(self):
...
         self.clear_info()
 
+    def _repo_id_generator(self):
+        """ Repository id numbers generator
+        """
+        repo_id = 0
+        while(True):
+            yield repo_id
+            repo_id += 1
+
```
Comment (vpodzime) [inline pyanaconda/ui/gui/spokes/source.py]: This should be a more generic thing not taking ``self`` and defined as ``iutil.id_generator``.

**Comment by jkonecny12** [comment]
Changed, thank you @vpodzime

**Comment by vpodzime** [comment]
Looks good to me now. Thanks for the extra work. ;)

**Comment by jkonecny12** [comment]
Pushed

**Pull request closed without merging**